pyswmm/links.py

```diff
@@ -0,0 +1,693 @@
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.offset1
+        >>> 0.1
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.offset1)
+
+    @offset1.setter
+    def offset1(self, param):
+        """
+        Set Link Upstream Link Offset
+
+        :param float param: New Parameter value
+  
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.offset1
+        ...     link.offset1 = 0.2
+        ...     print link.offset1
+        >>> 0.1
+        >>> 0.2
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.offset1, param)
+        
+    @property
+    def offset2(self):
+        """
+        Get Downstream Offset Depth
+
+        :return: Paramater Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.offset2
+        >>> 0.1
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.offset1)
+
+    @offset2.setter
+    def offset2(self, param):
+        """
+        Set Link Downstream Link Offset
+
+        :param float param: New Parameter value
+  
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.offset2
+        ...     link.offset2 = 0.2
+        ...     print link.offset2
+        >>> 0.1
+        >>> 0.2
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.offset2, param)
+    
+    @property
+    def q0(self):
+        """
+        Get Link Initial Flow
+
+        :return: Paramater Value
+        :rtype: float
+
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.q0
+        >>> 0
+        """
+        return self._model.getLinkParam(self._linkid,LinkParams.q0)
+
+    @q0.setter    
+    def q0(self, param):
+        """
+        Set Link Initial Flow Rate
+
+        :param float param: New Parameter value
+  
+        Examples:
+
+        >>> from pyswmm import Simulation
+        >>>
+        >>> with Simulation('../test/TestModel1_weirSetting.inp') as sim:
+        ...     link = Links(sim)["C1:C2"]
+        ...     print link.q0
+        ...     link.offset1 = 0.2
+        ...     print link.offset1
+        >>> 0.1
+        >>> 0.2
+        """  
+        self._model.setLinkParam(self._linkid,LinkParams.q0, param)
+    
+    @property
+    def qlimit(self):
```

Comment:
these names are weird :-|

limit.. q0 ... offset1 etc....

I think we can use much better names,